rtgui/editcallbacks.h

```diff
@@ -0,0 +1,256 @@
+    @param picked True if the cursor is still above the the same object than on button pressed and with the same modifier keys.
+                  If false, the user moved the cursor away or the modifier key is different, so the element is considered as NOT selected.
+    @return true if the preview has to be redrawn, false otherwise */
+    virtual bool pick2 (const bool picked);
+
+    /** @brief Triggered when the user is releasing mouse button 3 while in action==ES_ACTION_PICKING mode
+    @param picked True if the cursor is still above the the same object than on button pressed and with the same modifier keys.
+                  If false, the user moved the cursor away or the modifier key is different, so the element is considered as NOT selected.
+    @return true if the preview has to be redrawn, false otherwise */
+    virtual bool pick3 (const bool picked);
+
+    /** @brief Get the geometry to be shown to the user */
+    const std::vector<Geometry*>& getVisibleGeometry ();
+
+    /** @brief Get the geometry to be drawn in the "mouse over" channel, hidden from the user */
+    const std::vector<Geometry*>& getMouseOverGeometry ();
+};
+
+/** @brief Class to handle the furniture of data to the subscribers.
+ *
+ * It is admitted that only one Subscriber can ask data at a time. If the Subscriber is of type ET_PIPETTE, it will have to
+ * trigger the usual event so that the image will be reprocessed to compute the buffer of the current subscriber.
+ */
+class EditDataProvider
+{
+
+private:
+    EditSubscriber *currSubscriber;
+
+public:
+    int object;            /// ET_OBJECTS mode: Object detected under the cursor, 0 otherwise; ET_PIPETTE mode: 1 if above the image, 0 otherwise
+    float pipetteVal[3];   /// Current pipette values; if bufferType==BT_SINGLEPLANE_FLOAT, #2 & #3 will be set to 0
+
+    rtengine::Coord posScreen;       /// Location of the mouse button press, in preview image space
+    rtengine::Coord posImage;        /// Location of the mouse button press, in the full image space
+    rtengine::Coord deltaScreen;     /// Delta relative to posScreen
+    rtengine::Coord deltaImage;      /// Delta relative to posImage
+    rtengine::Coord deltaPrevScreen; /// Delta relative to the previous mouse location, in preview image space
+    rtengine::Coord deltaPrevImage;  /// Delta relative to the previous mouse location, in the full image space
+
+    EditDataProvider();
+    virtual ~EditDataProvider() {}
+
+    virtual void        subscribe(EditSubscriber *subscriber);
+    virtual void        unsubscribe();         /// Occurs when the subscriber has been switched off first
+    virtual void        switchOffEditMode ();  /// Occurs when the user want to stop the editing mode
+    virtual CursorShape getCursor(int objectID);
+    int                 getPipetteRectSize ();
+    EditSubscriber*     getCurrSubscriber();
+    virtual void        getImageSize (int &w, int&h) = 0;

```

Comment:
I'm always wondering why we don't have a `Size` or `Dimension` class for things like this. Or do we? And if we had/have why such a size could be negative.